packages/terra-button/src/variants/_IconButton.jsx

```diff
@@ -0,0 +1,320 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import classNamesBind from 'classnames/bind';
+import * as KeyCode from 'keycode-js';
+import ThemeContext from 'terra-theme-context';
+import styles from '../Button.module.scss';
+import { ButtonVariants, ButtonTypes, IconTypes } from '../_constants';
+
+const cx = classNamesBind.bind(styles);
+
+const propTypes = {
+  /**
+   * A icon to be displayed nested inline with the text when provided.
+   */
+  icon: PropTypes.element.isRequired,
+  /**
+   * specifies the icon type.
+   * refer Icon [Accessiblity Guide](https://engineering.cerner.com/terra-ui/components/cerner-terra-core-docs/icon/accessibility-guide) to know about types of icon.
+   */
+  iconType: PropTypes.oneOf([IconTypes.DECORATIVE, IconTypes.INFORMATIVE]).isRequired,
+  /**
+   * Whether or not the button should only display as an icon.
+   */
+  isIconOnly: PropTypes.bool,
+  /**
+   * Whether or not the button should display as a block.
+   */
+  isBlock: PropTypes.bool,
+  /**
+   * Whether or not the button has reduced padding
+   */
+  isCompact: PropTypes.bool,
+  /**
+   * Whether or not the button should be disabled.
+   */
+  isDisabled: PropTypes.bool,
+  /**
+   * Reverses the position of the icon and text.
+   */
+  isReversed: PropTypes.bool,
+  /**
+   * Callback function triggered when mouse is pressed.
+   */
+  onMouseDown: PropTypes.func,
+  /**
+   * Callback function triggered when clicked.
+   */
+  onClick: PropTypes.func,
+  /**
+   * Callback function triggered when button loses focus.
+   */
+  onBlur: PropTypes.func,
+  /**
+   * Callback function triggered when button gains focus.
+   */
+  onFocus: PropTypes.func,
+  /**
+   * Callback function triggered when key is pressed.
+   */
+  onKeyDown: PropTypes.func,
+  /**
+   * Callback function triggered when key is released.
+   */
+  onKeyUp: PropTypes.func,
+  /**
+   * Callback ref to pass into the dom element.
+   */
+  refCallback: PropTypes.func,
+  /**
+   * Sets the button text.
+   * If the button is `isIconOnly` or variant `utility` this text is set as the aria-label and title for accessibility.
+   */
+  text: PropTypes.string,
+  /**
+   * Additional information to display as a native tooltip on hover.
+   * Buttons declared as `isIconOnly` or `utility` will fallback to using `text` if not provided.
+   */
+  title: PropTypes.string,
+  /**
+   * Sets the button type. One of `button`, `submit`, or `reset`.
+   */
+  type: PropTypes.oneOf([ButtonTypes.BUTTON, ButtonTypes.SUBMIT, ButtonTypes.RESET]),
+  /**
+   * Sets the button variant. One of `action` or `utility`.
+   */
+  variant: PropTypes.oneOf([ButtonVariants.ACTION, ButtonVariants.UTILITY]),
```

Comment:
Should this list also contain neutral as default variant is neutral. Or should the default variant be one of action/utility?

Reply:
I missed adding back the variants. IconButton supports all the variants. 
Added [here](https://github.com/cerner/terra-core/pull/3640/commits/e98c61fd2be7dc305b360c947a5a11a5be6358de)